builder/build-image.sh

```diff
@@ -37,22 +37,39 @@ apt-get update && apt-get install -y packer
 packer plugins install github.com/hashicorp/qemu
 packer plugins install github.com/hashicorp/ansible
 
-# do the build
-if tools/build.sh "$IMAGE" >/var/log/image-build.log 2>&1; then
-    # compress the resulting image
-    QEMU_SOURCE_ID=$(hcl2tojson "$IMAGE" | jq -r '.source[0].qemu | keys[]')
-    VM_NAME=$(hcl2tojson "$IMAGE" | jq -r '.source[0].qemu.'"$QEMU_SOURCE_ID"'.vm_name')
-    QCOW_FILE="$VM_NAME.qcow2"
-    builder/refresh.sh vo.access.egi.eu "$(cat /var/tmp/egi/.refresh_token)" images
-    OS_TOKEN="$(yq -r '.clouds.images.auth.token' /etc/openstack/clouds.yaml)"
-    OUTPUT_DIR="$(dirname "$IMAGE")/output-$QEMU_SOURCE_ID"
-    cd "$OUTPUT_DIR"
-    qemu-img convert -O qcow2 -c "$VM_NAME" "$QCOW_FILE"
-    openstack --os-cloud images --os-token "$OS_TOKEN" \
-        object create egi_endorsed_vas "$QCOW_FILE"
-    ls -lh "$QCOW_FILE"
-    SHA="$(sha512sum -z "$QCOW_FILE" | cut -f1 -d" ")"
-    echo "SUCCESSFUL BUILD - $QCOW_FILE - $SHA" >>/var/log/image-build.log
+QEMU_SOURCE_ID=$(hcl2tojson "$IMAGE" | jq -r '.source[0].qemu | keys[]')
+VM_NAME=$(hcl2tojson "$IMAGE" \
+	| jq -r '.source[0].qemu.'"$QEMU_SOURCE_ID"'.vm_name')
+QCOW_FILE="$VM_NAME.qcow2"
+
+# Check if the image is already there
+builder/refresh.sh vo.access.egi.eu "$(cat /var/tmp/egi/.refresh_token)" images
+OS_TOKEN="$(yq -r '.clouds.images.auth.token' /etc/openstack/clouds.yaml)"
+if openstack --os-cloud images --os-token "$OS_TOKEN" \
+	object show egi_endorsed_vas \
+	"$QCOW_FILE"  > /dev/null ; then
+	# skip
+	echo "Skipped build as image is already uploaded" >>/var/log/image-build.log
+else
+	# do the build
+	{
+		if tools/build.sh "$IMAGE"; then
+			# refresh the token, it may have expired
+			OUTPUT_DIR="$(dirname "$IMAGE")/output-$QEMU_SOURCE_ID"
+			cd "$OUTPUT_DIR"
+			# compress the resulting image
+			qemu-img convert -O qcow2 -c "$VM_NAME" "$QCOW_FILE"
+			builder/refresh.sh vo.access.egi.eu \
+				"$(cat /var/tmp/egi/.refresh_token)" images
+			OS_TOKEN="$(yq -r '.clouds.images.auth.token' \
+				/etc/openstack/clouds.yaml)"
+			openstack --os-cloud images --os-token "$OS_TOKEN" \
+				object create egi_endorsed_vas \
+				"$QCOW_FILE"
+			SHA="$(sha512sum -z "$QCOW_FILE" | cut -f1 -d" ")"
+			echo "SUCCESSFUL BUILD - $QCOW_FILE - $SHA"
+		fi
+	} >>/var/log/image-build.log 2>&1
```

Comment:
With this:
https://github.com/EGI-Federation/fedcloud-vmi-templates/blob/f36bd71c2daab5fbcdab513c3d58c8a8bff4940d/builder/build-image.sh#L4-L12

I think we can remove the redirection to the log, and move the code inside curly brackets to the left.

Reply:
Do you prefer to keep the block above anyway? Just confirming before I approve.

Reply:
agree, I'm modifying